File: HackerRank/Algorithms/3. Strings/Alternating_characters.java
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here.
           Read input from STDIN.
           Print output to STDOUT.
           Your class should be named Solution. */
        Scanner scan = new Scanner(System.in);
        int testCase;
        testCase = scan.nextInt();
        String str;
        int i, j, deleteCharactersCount;
        scan.nextLine();
        while (testCase != 0) {
            str = scan.nextLine();
            deleteCharactersCount = 0;
            i = 1;
            j = 0;
            while (i < str.length()) {
                if (str.charAt(i) == str.charAt(j)) {
                    deleteCharactersCount++;
                    ++i;
                }
                else {
                    j = i;
                    ++i;
                }
            }
            System.out.println(deleteCharactersCount);
            testCase--;
        }
        scan.close();
    }
}

File: HackerRank/Algorithms/3. Strings/Alternating_characters.cpp
#include <iostream>
using namespace std;

int main() {
    /* Enter your code here.
       Read input from STDIN.
       Print output to STDOUT */
    int test_case;
    string input;
    int i, j;
    ios::sync_with_stdio(false);
    cin >> test_case;
    while (test_case--) {
        cin >> input;
        int delete_characters_count = 0;
        i = 1;
        j = 0;
        while (i < input.length()) {
            if (input[i] == input[j]) {
                delete_characters_count++;
                ++i;
            }
            else {
                j = i;
                ++i;
            }
        }
        cout << delete_characters_count << endl;
    }
    return 0;
}
